src/sys/signal.rs

```diff
@@ -1161,6 +1156,14 @@ pub enum SigevNotify {
         /// structure of the queued signal.
         si_value: libc::intptr_t
     },
+    /// A helper variant to resolve the unused parameter (`'fd`) problem on
+    /// platforms other than FreeBSD and DragonFlyBSD.
+    ///
+    /// This variant can never be constructed due to the usage of an enum with 0
+    /// variants.
+    #[doc(hidden)]
+    #[cfg(not(freebsdlike))]
+    _Unreachable(&'fd std::convert::Infallible),
```

Comment:
I changed the associated type of this variant to `&'fd Infallible`, since `Infallible` is an enum with 0 variants, this variant can now never be constructed. (As a contrast, `PhantomData` is very easy to create)